caches/clan_test.go

```diff
@@ -0,0 +1,125 @@
+package caches_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	uuid "github.com/satori/go.uuid"
+	"github.com/topfreegames/khan/models"
+	"github.com/topfreegames/khan/testing"
+)
+
+var _ = Describe("Clan Cache", func() {
+	var testDb models.DB
+
+	BeforeEach(func() {
+		var err error
+		testDb, err = testing.GetTestDB()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Describe("Clans Summaries", func() {
+		getPublicIDsAndIDToIndexMap := func(clans []*models.Clan) ([]string, map[string]int) {
+			var publicIDs []string
+			idToIdx := make(map[string]int)
+			for i, clan := range clans {
+				publicIDs = append(publicIDs, clan.PublicID)
+				idToIdx[clan.PublicID] = i + 1
+			}
+			return publicIDs, idToIdx
+		}
+
+		assertFirstCacheCall := func(clans []*models.Clan, idToIdx map[string]int, clansSummaries []map[string]interface{}) {
```

Comment:
Refactor this into a top-level function?

Ditto for assertSecondCacheCall.

Reply:
Why? This test file may cover more caches, I think these functions are better declared within the scope of the specific cache they're making assertions about.